### ISSUE TITLE
Cannot read property 'applicationData' of undefined

### ISSUE THREAD

**Issue by runjiezhang** [F5Networks/f5-declarative-onboarding#204]
### Environment
 * Declarative Onboarding Version: 1.18.0-4
 * BIG-IP Version: 15.0.1.3

### Summary
When attempting to config the bigip 15.0.1.3 using DO 1.18.0-4, the processing of the DO declaration is failing due to the error message: `"Cannot read property 'applicationData' of undefined"` even though the Disk section is not included.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
         "class": "Tenant",
            "configsync": {
                "class": "ConfigSync",
                "configsyncIp": "10.112.90.90"
            },
            "default": {
                "class": "ManagementRoute",
                "gw": "10.255.88.1",
                "mtu": 1500,
                "network": "default"
            },
            "external": {
                "address": "10.96.90.90/12",
                "allowService": "none",
                "class": "SelfIp",
                "trafficGroup": "traffic-group-local-only",
                "vlan": "external_vlan"
            },
            "external_vlan": {
                "class": "VLAN",
                "interfaces": [
                    {
                        "name": "1.2"
                    }
                ],
                "mtu": "1500"
            },
            "failoverAddress": {
                "address": "10.112.90.90",
                "class": "FailoverUnicast"
            },
            "internal": {
                "address": "10.112.90.90/12",
                "allowService": "default",
                "class": "SelfIp",
                "trafficGroup": "traffic-group-local-only",
                "vlan": "internal_vlan"
            },
            "internal2": {
                "address": "10.128.90.90/12",
                "allowService": "default",
                "class": "SelfIp",
                "trafficGroup": "traffic-group-local-only",
                "vlan": "internal2_vlan"
            },
            "internal_vlan": {
                "class": "VLAN",
                "interfaces": [
                    {
                        "name": "1.1"
                    }
                ],
                "mtu": "1500"
            },
            "myDns": {
                "class": "DNS",
                "nameServers": [
                    "172.27.1.1"
                ],
                "search": [
                    "olympus.f5net.com",
                    "f5net.com"
                ]
            },
            "myLicense": {
                "class": "License",
                "licenseType": "regKey",
                "overwrite": false,
                "regKey": "U3100-71853-80092-96021-7040367"
            },
            "myNtp": {
                "class": "NTP",
                "servers": [
                    "ntp.f5net.com"
                ],
                "timezone": "America/Los_Angeles"
            },
            "myProvisioning": {
                "afm": "minimum",
                "asm": "minimum",
                "avr": "minimum",
                "class": "Provision",
                "ltm": "nominal"
            },
            "mySystem": {
                "class": "System",
                "hostname": "runzhang-bigip-1.olympus.f5net.com"
            },
            "trust": {
                "class": "DeviceTrust",
                "localPassword": "admin",
                "localUsername": "admin",
                "remoteHost": "10.255.90.90",
                "remotePassword": "admin",
                "remoteUsername": "admin"
            }
        },
        "async": true,
        "class": "Device",
        "label": "bigip1 declaration for declarative onboarding",
        "schemaVersion": "1.18.0"
}
```

2. Observe the following error response:
```json
{
   "id\":\"0eca797d-c56d-4b97-bd7e-d2175e89ac4e\",\"selfLink\":\"https://localhost/mgmt/shared/declarative-onboarding/task/0eca797d-c56d-4b97-bd7e-d2175e89ac4e\",\"code\":500,\"status\":\"ERROR\",\"message\":\"failed\",\"errors\":[\"Cannot read property 'applicationData' of undefined\",\"Cannot read property 'applicationData' of undefined\",\"Cannot read property 'applicationData' of undefined\"],\"result\":{\"class\":\"Result\",\"code\":500,\"status\":\"ERROR\",\"message\":\"failed\",\"errors\":[\"Cannot read property 'applicationData' of undefined\",\"Cannot read property 'applicationData' of undefined\",\"Cannot read property 'applicationData' of undefined\"]}
    "declaration": {}
}
```

### Expected Behavior
The DO configuration should finish successfully. 

### Actual Behavior
The process failed with the error: `"Cannot read property 'applicationData' of undefined"`.
Response URL: https://10.255.90.90/mgmt/shared/declarative-onboarding/task/0eca797d-c56d-4b97-bd7e-d2175e89ac4e

**Comment by dstokesf5** [comment]
Are you seeing this on your first DO post to a BIG-IP, or are you updating a DO declaration? Also, are you able to simplify the declaration at all and still reproduce the error?

**Comment by runjiezhang** [comment]
> Are you seeing this on your first DO post to a BIG-IP, or are you updating a DO declaration? Also, are you able to simplify the declaration at all and still reproduce the error?

I am seeing this on my first DO post. I can still reproduce it after I simplify the declaration to
```json
{
  "Common": {
            "class": "Tenant",
            "myDns": {
                "class": "DNS",
                "nameServers": [
                    "172.27.1.1"
                ],
                "search": [
                    "olympus.f5net.com",
                    "f5net.com"
                ]
            },
            "myLicense": {
                "class": "License",
                "licenseType": "regKey",
                "overwrite": false,
                "regKey": "U3100-71853-80092-96021-7040367"
            },
            "myNtp": {
                "class": "NTP",
                "servers": [
                    "ntp.f5net.com"
                ],
                "timezone": "America/Los_Angeles"
            },
            "myProvisioning": {
                "afm": "minimum",
                "asm": "minimum",
                "avr": "minimum",
                "class": "Provision",
                "ltm": "nominal"
            },
            "mySystem": {
                "class": "System",
                "hostname": "runzhang-bigip-1.olympus.f5net.com"
            }
        },
        "async": true,
        "class": "Device",
        "label": "bigip1 declaration for declarative onboarding",
        "schemaVersion": "1.18.0"
}
```

**Comment by dstokesf5** [comment]
Thank you. This information will be very helpful in reproducing and triaging this issue.

**Comment by runjiezhang** [comment]
> Thank you. This information will be very helpful in reproducing and triaging this issue.

No problem. I even tried with the following declaration and got the same error:

```json
 {
        "async": true,
        "class": "Device",
        "label": "bigip1 declaration for declarative onboarding",
        "schemaVersion": "1.18.0"
}
```

**Comment by dstokesf5** [comment]
Looks like this is actually pretty easy to reproduce by submitting a second declaration. I have added it to our internal product backlog as AUTOTOOL-2242.

**Comment by dstokesf5** [comment]
This issue was resolved in the 1.20.0 release.
